src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -859,5 +846,39 @@ protected override void Dispose(bool disposing)
             var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(false);
             await props.ActiveDebugProfile.SetValueAsync(profileName).ConfigureAwait(true);
         }
+
+        internal async Task<string> GetLaunchSettingsFileAsync()
+        {
+            if (_launchSettingsFile == null)
+            {
+                _launchSettingsFile = Path.Combine(await GetLaunchSettingsFileFolderAsync(), LaunchSettingsFilename);
+            }
+
+            return _launchSettingsFile;
+        }
+
+        private async Task<string> GetLaunchSettingsFileFolderAsync(bool createIfNotExists = false)
+        {
+            // NOTE: To reduce behavior changes, we currently cache the folder that we get from
+            // the ISpecialFilesManager, even though it can change over the lifetime of the project
+            // We should fix this, see: https://github.com/dotnet/project-system/issues/2316.
+
+            if (_launchSettingsFileFolder == null)
```

Comment:
You cached the value, which means that we don't expect the customer to edit the project file and change it during the session.  I guess it is a known limitation, and we can live with it, right?